README.md

```diff
@@ -1,73 +1,218 @@
 
-Have the views you want to demo conform to `Demoable`:
 ```swift
-extension AwesomeViewController: Demoable {
-    var title: String {
-        "Awesome!"
+import UIKit
+import DemoKit
+
+class MySimpleDemoView: UIView, Demoable {
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        backgroundColor = .systemBlue
     }
 
-    var dismissKind: DismissKind {
-        .button // Overlays a button to dismiss the view
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
     }
 }
 ```
 
-Add the views to a `DemoGroup`:
+### Group demos using the `DemoGroup` protocol
+The [`DemoGroup`](https://github.com/finn-no/DemoKit-iOS/blob/main/Sources/DemoKit/Public/Protocols/DemoGroup.swift) protocol is used to organize and categorize your demos into logical groups. Each group represents a collection of related demos, making it easier for users to navigate and explore them in your demo app. However, there's no shame in only using a single group for all your demos.
+
+#### What is `DemoGroup`?
+A `DemoGroup` defines a set of demos and how they are presented in the app. It’s typically implemented as an enum, where each case corresponds to a specific demo. This makes the implementation lightweight, easy to maintain, and expressive.
+
+When implementing a `DemoGroup` as an enum with `String` as the raw value, certain default behaviors are provided to make setup even simpler. This approach leverages the raw value to automatically generate key properties like titles.
+
+#### Key properties
+
+| Property | Type | Description |
+| - | - | - |
+| `numberOfDemos` | `Int` | The total number of demos in the group. |
+| `groupTitle` | `String` | The title displayed for the group in the navigation bar or group selector.<br>Default value: A stringified version of the `DemoGroup` type. |
+
+#### Key methods
+
+For the methods that take `index` as a parameter this value refers to the index of the demo, and will always be between `0` and `numberOfDemos - 1`.
+
+| Method | Return type | Description |
+| - | - | - |
+| `demoGroupItem(for index: Int)` | [`DemoGroupItem`](https://github.com/finn-no/DemoKit-iOS/blob/main/Sources/DemoKit/Public/Protocols/DemoGroupItem.swift) | Returns the metadata (e.g., title, subtitle) for a demo at the specified index. This is used to populate the list of demos for this group. |
+| `demoable(for index: Int)` | [`Demoable`](https://github.com/finn-no/DemoKit-iOS/blob/main/Sources/DemoKit/Public/Protocols/Demoable.swift) | Returns the actual demo (`Demoable`) at the specified index. This is invoked when a user selects a demo from the list. |
+
+#### Example usage
+
 ```swift
-enum ViewsDemoGroup: String, CaseIterable, DemoGroup, DemoGroupItem {
-    case awesomeView
-    case superbView
+import UIKit
+import DemoKit
+
+// An enum, where all cases refers to a specific demo in your app.
+// Using a 
```

Comment:
Seems like an uncompleted phrase

Reply:
Haha it definitely does😄 I'll see if I can pick up where I left off